Docker Compose for whole system

seanmcilvenna opened the pull request:
* docker-compose.yml that has all services needed to run the entire stack of services
* supporting services include:
  * kafka broker
  * kafka ui
  * sql server
  * mongo
  * postgres
  * redis
  * loki
  * prometheus
  * grafana
* init containers;
  * broker has init that creates topics
  * sql database has a `create_dbs.sql` file that creates the dbs
* run `docker compose build` from root to build all services
* run `docker compose up` to run all services
* each service declares dependencies
* occasional modifications to default app settings
* a new appsettings.Docker.json file that is used by the docker-compose to launch the services with a specific port (on http, not https)
* modified (and created) .dockerignore files to exclude built artifacts (new .dockerignores needed at the root context of each build)
* some changes made to the Java services to have them build and run successfully in the docker compose
* consistency across all projects for naming conventions like the name of the app in logs "link-cloud"
* all services aggregate logs to loki and can be accessed via grafana
* deleted project folders that are no longer used and no longer relevant

seanmcilvenna committed:
Removing service folders that are no longer used.
`docker-compose` now builds all services, but still need to get all the services to run together

seanmcilvenna committed:
Account, Audit and Census services run successfully in docker-compose

seanmcilvenna committed:
Tenant running now.
Checked that Loki is working with services thus far.

seanmcilvenna committed:
Updating logging to use "link-cloud" as the "app" property for all services

seanmcilvenna committed:
Reverting change to remove Domain project from project references after re-basing

seanmcilvenna committed:
Aligning Serilog to use the same order of sinks across services
Worked on normalization docker compose

seanmcilvenna committed:
Merge branch 'dev' into docker-compose-improvements

# Conflicts:
#	DotNet/Census/Program.cs
#	DotNet/DataAcquisition/appsettings.json

seanmcilvenna committed:
Removing unused constans from DA
Renaming the SQL retry repo to use consistent naming pattern for census

seanmcilvenna committed:
Removing unused docker-compose.override.yml
Updating comment to reflect "database" rather than "mongo"

seanmcilvenna committed:
Fixing topic names for error and retry
Removing extra constant now that the PR has been accepted in dev branch

seanmcilvenna committed:
changing dependencies to ensure that when services run, that kafka_init has run to make sure a topic has been created for it
correcting census and data acqusition serilog default configs

seanmcilvenna committed:
Updating for PatientsToQuery for docker-compose

seanmcilvenna committed:
no message

seanmcilvenna committed:
Re-adding ENV lines for Entity Framework

seanmcilvenna committed:
Updating pipelines to only trigger for PRs that modify the service's code
Removing the "web" pipelines that are duplicate to "demo"
Removing the "shared" pipelines because all the pipelines build "shared" as part of each service

seanmcilvenna committed:
Patient list pipelines no longer valid. Duplicates of Census

seanmcilvenna committed:
Merge branch 'docker-dependencies' into docker-compose-improvements

# Conflicts:
#	DotNet/Census/Dockerfile

seanmcilvenna committed:
Updating pipelines to build Docker images from a context of the `Java` directory to reduce the amount of files needing to be copied into docker context for java projects
Updating measure eval pipelines to build Java

seanmcilvenna committed:
Updating pipelines to build Docker images from a context of the `Java` directory to reduce the amount of files needing to be copied into docker context for java projects
Updating measure eval pipelines to build Java

(cherry picked from commit 039c1f617f5a1c74f948e6b3c81e867d017ca149)

smailliwcs committed:
Fix `measureeval` Dockerfile

Expect a build context of Java rather than Java/measureeval.

seanmcilvenna committed:
Merge branch 'fix-measureeval-dockerfile' into docker-dependencies

seanmcilvenna committed:
Merge branch 'docker-dependencies' into docker-compose-improvements

seanmcilvenna committed:
Merge branch 'dev' into docker-compose-improvements

seanmcilvenna committed:
Updating measureeval

seanmcilvenna committed:
Merge branch 'dev' into docker-dependencies

seanmcilvenna committed:
Merge branch 'docker-dependencies' into docker-compose-improvements

seanmcilvenna committed:
Normalization and Java MeasureEval service working

seanmcilvenna committed:
Merge branch 'dev' into docker-compose-improvements

seanmcilvenna committed:
Merge branch 'dev' into docker-compose-improvements

# Conflicts:
#	DotNet/Census/appsettings.DockerLocal.json
#	DotNet/Census/appsettings.docker.json

seanmcilvenna committed:
Enabling swagger in all Docker app configs
Using ubuntu base image for Notification
Notification, QueryDispatch, PatientsToQuery, Report and Submission are all running in docker-compose

seanmcilvenna committed:
Adding .Domain into consideration for the CI and Dockerfile

seanmcilvenna committed:
Basing Normalization service on Ubuntu image to handle invariant culture issue

smailliwcs committed:
Upgrade Spring Boot

seanmcilvenna committed:
Merge branch 'docker-dependencies' into docker-compose-improvements

smailliwcs committed:
Version all modules to 0.1.0-SNAPSHOT

smailliwcs committed:
Delegate to `spring-boot-dependencies`

Where possible, allow dependency versions to flow in from ancestor POMs,
rather than explicitly fixing them in our root POM.

smailliwcs committed:
Remove duplicate property declarations

Java version (17) and source encoding (UTF-8) should both flow in from
`spring-boot-starter-parent`.

seanmcilvenna committed:
Updating docker ignore files to reduce build context

seanmcilvenna committed:
Updating version of Confluent.Kafka used by BFF so that there isn't a conflict with Shared's dependency version

seanmcilvenna committed:
Merge branch 'dev' into docker-compose-improvements

# Conflicts:
#	DotNet/Normalization/Application/Settings/NormalizationConstants.cs
#	DotNet/Normalization/appsettings.Development.json
#	DotNet/Notification/Dockerfile
#	Java/measureeval/Dockerfile
#	Java/validation/Dockerfile

seanmcilvenna committed:
Fixed data acquisition to use MSSQL instead of mongo

seanmcilvenna committed:
Fixing submission's log aggregation to loki

seanmcilvenna committed:
Fixing Tenant's log aggregation to loki

smailliwcs committed:
Fix `validation` initialization (#163)

- Replace Mongo starter dependency
  - Use Spring Boot plugin (in module POM only)
  - Enable JPA repositories

seanmcilvenna committed:
Validation runs in docker-compose now

seanmcilvenna committed:
Merge branch 'dev' into docker-compose-improvements

smailliwcs committed:
Move Spring Boot plugin to module POM

edward-miller-lcg reviewed (approved):
Would like to see @dvargaslantana or @amphillipsLGC or @smailliwcs provide an approval as well in case I missed something. But overall looks good to me.

smailliwcs committed:
Downgrade HAPI to 7.0.0

This works around an issue with the latest `cqf-fhir` libraries:

https://github.com/cqframework/clinical-reasoning/issues/451

For now, use version 3.0.0 of those libraries, which depend transitively
on HAPI 7.0.0 (which we declare explicitly for compatibility).

smailliwcs committed:
Declare dependencies in module POMs

smailliwcs committed:
Fix health check route

smailliwcs committed:
Integrate `measureeval` with `main`

smailliwcs committed:
Reformat POMs for consistency

smailliwcs committed:
Merge branch 'docker-compose-improvements' into java-deployment

smailliwcs committed:
Make Docker builds consistent

This includes a change to the `validation` parent image from OpenJDK
(which is deprecated) to Eclipse Temurin.

smailliwcs committed:
Add validation packages

These were previously being ignored due to a rule in the root-level
.gitignore.  This commit inverts that rule (specifically in the Java
resources directory for `validation`) and copies in the set of packages
currently being used in MVP.

smailliwcs committed:
Add missing dependency

smailliwcs committed:
Fix port and issuer URI

smailliwcs commented on .env:
```diff
@@ -0,0 +1,3 @@
+LINK_DB_USER=link-db-user
+LINK_DB_PASS=link-db-pass
+REDIS_PASS=suT1ChwYtkwWfbV1bUfRvQZO1YNhNRMj
```
Is this sensitive?

seanmcilvenna replied:
No. This is a new uniquely generated value just for this docker-compose.

smailliwcs reviewed (approved):
Approved, and tacked on further Java-specific changes in a `java-deployment` branch.

seanmcilvenna committed:
Merge pull request #167 from lantanagroup/java-deployment

Make Java deployments consistent

seanmcilvenna committed:
Updating java validation to use correct syntax for running with `--add-opens` options
Adding system `-add-options` option to measure eval service dockerfile

seanmcilvenna committed:
Gettinng BFF to work... A bit more work to do

seanmcilvenna committed:
Merge branch 'dev' into docker-compose-improvements

# Conflicts:
#	DotNet/DataAcquisition/appsettings.Development.json
